CSharpSourceCode/Abilities/Crosshairs/TargetedAOECrosshair.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using TaleWorlds.Engine;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace TOW_Core.Abilities.Crosshairs
+{
+    public class TargetedAOECrosshair : AbilityCrosshair
+    {
+        public TargetedAOECrosshair(AbilityTemplate template) : base(template)
+        {
+            _crosshair.EntityFlags |= EntityFlags.NotAffectedBySeason;
+            UpdateFrame();
+            IsVisible = false;
+        }
+        public override void Tick()
+        {
+            UpdateFrame();
+            if (CollidedAgents != null)
+            {
+                previousAgents = (Agent[])CollidedAgents.Clone();
```

Comment:
what exactly does here clone do?

Reply:
i need to get copy of CollidedAgents array, not reference. Maybe it's excess, idk